Makefile

```diff
@@ -48,25 +49,44 @@ generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and
 
 .PHONY: fmt
 fmt: ## Run go fmt against code.
-	go fmt ./...
+	@echo
+	@echo "===> Formatting Go files <==="
+	@gofmt -s -l -w $(GO_FILES)
 
 .PHONY: vet
 vet: ## Run go vet against code.
 	go vet ./...
 
+$(GOLANGCI_LINT_BIN):
+	@echo "===> Installing Golangci-lint <==="
+	@rm -rf $(GOLANGCI_LINT_BINDIR)/* # remove old versions
+	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOLANGCI_LINT_BINDIR)/$(GOLANGCI_LINT_VERSION) $(GOLANGCI_LINT_VERSION)
+
+.PHONY: golangci
+golangci: $(GOLANGCI_LINT_BIN)
+	@echo "===> Running golangci (linux) <==="
+	@GOOS=linux $(GOLANGCI_LINT_BIN) run -c $(CURDIR)/.golangci.yml
+
+.PHONY: golangci-fix
+golangci-fix: $(GOLANGCI_LINT_BIN)
+	@echo "===> Running golangci (linux) <==="
+	@GOOS=linux $(GOLANGCI_LINT_BIN) run -c $(CURDIR)/.golangci.yml --fix
+
+.PHONY: .coverage
+.coverage:
+	mkdir -p $(CURDIR)/.coverage
+
 .PHONY: test
-test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test -gcflags=all=-l ./... -coverprofile cover.out  ## Prohibit inline optimization when using gomonkey
+test: manifests generate fmt vet envtest .coverage## Run tests .
```

Comment:
`.coverage## Run tests .` can work but I think we'd better change it to `.coverage  ## Run tests.`

Reply:
thanks， changed